client/packages/sdk/src/price-estimation/__tests__/eth.test.ts

```diff
@@ -0,0 +1,99 @@
+import { describe, expect, it } from "@jest/globals";
+import fetch from "node-fetch";
+import {
+  SpeedModes,
+  estimateGasFee,
+  getGasPrice,
+  getGasAmount,
+  Actions,
+  ETH_TRANSFER_GAS_AMOUNT,
+} from "../eth";
+
+jest.mock("node-fetch-commonjs", () => jest.fn());
+
+const mockGetGasPriceSuccesfulResponse = () => {
+  // @ts-ignore - mockImplementationOnce is not defined in type
+  fetch.mockImplementationOnce(() =>
+    Promise.resolve({
+      status: 200,
+      async json() {
+        return {
+          code: 200,
+          data: {
+            rapid: 17796706627,
+            fast: 14000849292,
+            standard: 13969553812,
+            slow: 13969553812,
+            timestamp: 1687612735879,
+            priceUSD: 1898.3793,
+          },
+        };
+      },
+    })
+  );
+};
+
+const mockGetGasPriceErrorResponse = () => {
+  // @ts-ignore - mockImplementationOnce is not defined in type
+  fetch.mockImplementationOnce(() =>
+    Promise.resolve({
+      status: 500,
+    })
+  );
+};
+
+describe("eth", () => {
+  describe("getGasPrice", () => {
+    it("should return the current gas price of ETH", async () => {
+      mockGetGasPriceSuccesfulResponse();
+
+      const result = await getGasPrice("eth");
+
+      expect(result).not.toEqual(undefined);
+      expect(result.rapid).toEqual(17796706627);
+      expect(result.fast).toEqual(14000849292);
+      expect(result.standard).toEqual(13969553812);
+      expect(result.slow).toEqual(13969553812);
+    });
+
+    it("should throw error if unable to get current ETH gas price", async () => {
+      mockGetGasPriceErrorResponse();
+
+      try {
+        await getGasPrice('eth');
+      } catch (e) {
+        expect(e).toEqual(
+          new Error("Failed to fetch gas price. ERROR_STATUS: 500")
+        );
+      }
+    });
+  });
+
+  describe("estimateGasFee", () => {
+    it("should get the estimate gas fee for a transfer", async () => {
```

Comment:
Please test all speed modes

Reply:
Fixed, refer to commit [0fd588d](https://github.com/t3rn/t3rn/pull/1091/commits/0fd588dfbcb1b371e9b60d1415e20932aa3e1ab4).